mail/mailer_test.go

```diff
@@ -74,17 +73,26 @@ func expect(t *testing.T, buf *bufio.Reader, expected string) error {
 	return nil
 }
 
-type connHandler func(int, *testing.T, net.Conn)
+type connHandler func(int, *testing.T, net.Conn, *net.TCPConn)
 
-func listenForever(l net.Listener, t *testing.T, handler connHandler) {
+func listenForever(l *net.TCPListener, t *testing.T, handler connHandler) {
 	connID := 0
 	for {
-		conn, err := l.Accept()
+		tcpConn, err := l.AcceptTCP()
 		if err != nil {
 			return
 		}
+
+		keyPair, err := tls.LoadX509KeyPair("../test/mail-test-srv/localhost/cert.pem", "../test/mail-test-srv/localhost/key.pem")
```

Comment:
Let's load the keypair once, before the loop. And for that matter, we can construct a single `tls.Config` and reuse it.